tbot_contrib/gpio.py

```diff
@@ -0,0 +1,56 @@
+import tbot
+from tbot.machine import linux
+
+
+class Gpio:
+    def __init__(self, host: linux.LinuxShell, gpio_number: int):
+
+        self.host = host
+        self._gpio_number = str(gpio_number)
+        self._gpio_sysclass_path = self.host.fsroot / "sys/class/gpio"
+        self._gpio_path = self._gpio_sysclass_path / f"gpio{self._gpio_number}"
+        self._export()
+
+    def _export(self) -> None:
+        if not self._gpio_path.is_dir():
+            self.host.exec0(
+                "echo",
+                str(self._gpio_number),
+                linux.RedirStdout(self._gpio_sysclass_path / "export"),
+            )
+
+    def _is_in_direction(self) -> bool:
+        direction = self.host.exec0("cat", self._gpio_path / "direction")
+        return direction.strip().lower() == "in"
```

Comment:
I think it makes more sense to store the current direction in an instance attribute:

```python
def __init__(self, ...):
	...

	self.direction = self.get_direction()

def set_direction(self, direction: str):
	(self._gpio_path / "direction").write_text(direction)
    self.direction = direction
```

and then just check that instead of running a command each time:

```python
def set_value(self, value: bool):
	if self.direction != "out":
	    raise Exception("Can't set a GPIO which is not an output")
    ...
```